File: .claude/examples/model_switcher.js
/**
 * AI 模型切换使用示例 (Node.js)
 * 支持本地 cursorweb2api 的 24 个模型
 */

const axios = require('axios');

class MultiModelAI {
    constructor(baseURL = 'http://localhost:8000/v1', apiKey = 'aaa') {
        this.baseURL = baseURL;
        this.apiKey = apiKey;
        this.client = axios.create({
            baseURL: this.baseURL,
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `Bearer ${this.apiKey}`
            }
        });
    }

    /**
     * 调用指定模型进行对话
     * @param {string} model - 模型ID
     * @param {string} message - 用户消息
     * @param {Object} options - 其他参数
     */
    async chat(model, message, options = {}) {
        const data = {
            model,
            messages: [{ role: 'user', content: message }],
            ...options
        };

        try {
            const response = await this.client.post('/chat/completions', data);
            return response.data.choices[0].message.content;
        } catch (error) {
            console.error(`错误 [${model}]:`, error.message);
            return null;
        }
    }

    /**
     * 列出所有可用模型
     */
    async listModels() {
        try {
            const response = await this.client.get('/models');
            return response.data.data.map(m => m.id);
        } catch (error) {
            console.error('获取模型列表失败:', error.message);
            return [];
        }
    }

    /**
     * 流式对话（实时输出）
     */
    async chatStream(model, message, onChunk) {
        const data = {
            model,
            messages: [{ role: 'user', content: message }],
            stream: true
        };

        try {
            const response = await this.client.post('/chat/completions', data, {
                responseType: 'stream'
            });

            response.data.on('data', chunk => {
                const lines = chunk.toString().split('\n').filter(line => line.trim());
                for (const line of lines) {
                    if (line.startsWith('data: ')) {
                        const json = line.slice(6);
                        if (json !== '[DONE]') {
                            try {
                                const parsed = JSON.parse(json);
                                const content = parsed.choices[0]?.delta?.content;
                                if (content) onChunk(content);
                            } catch (e) {}
                        }
                    }
                }
            });
        } catch (error) {
            console.error('流式请求失败:', error.message);
        }
    }
}

// 使用示例
async function main() {
    const ai = new MultiModelAI();

    console.log('📋 可用模型:');
    const models = await ai.listModels();
    models.forEach((model, i) => {
        console.log(`  ${i + 1}. ${model}`);
    });

    console.log('\n' + '='.repeat(50) + '\n');

    // 测试不同模型
    const testModels = [
        { id: 'claude-4.5-sonnet', name: 'Claude 4.5 Sonnet' },
        { id: 'gpt-4o', name: 'GPT-4o' },
        { id: 'gemini-2.5-flash', name: 'Gemini 2.5 Flash' },
        { id: 'deepseek-r1', name: 'DeepSeek R1' }
    ];

    const question = '什么是递归？用一句话解释。';

    console.log(`❓ 问题: "${question}"\n`);

    for (const { id, name } of testModels) {
        console.log(`🤖 ${name}:`);
        const answer = await ai.chat(id, question, { max_tokens: 100 });
        console.log(`   ${answer}\n`);
    }

    // 编程任务示例
    console.log('='.repeat(50));
    console.log('💻 编程任务测试\n');

    const codeQuestion = '写一个JavaScript函数，计算数组的平均值';
    console.log(`❓ ${codeQuestion}\n`);

    console.log('🤖 Code Supernova:');
    const codeAnswer = await ai.chat('code-supernova-1-million', codeQuestion, { max_tokens: 500 });
    console.log(codeAnswer);
}

// 交互式切换示例
async function interactiveMode() {
    const ai = new MultiModelAI();
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    console.log('🎯 交互式 AI 模型切换');
    console.log('命令: /model <模型名> - 切换模型');
    console.log('命令: /list - 列出所有模型');
    console.log('命令: /exit - 退出\n');

    let currentModel = 'claude-4.5-sonnet';
    console.log(`当前模型: ${currentModel}\n`);

    const askQuestion = () => {
        rl.question('You: ', async (input) => {
            input = input.trim();

            if (input === '/exit') {
                console.log('再见！');
                rl.close();
                return;
            }

            if (input === '/list') {
                const models = await ai.listModels();
                console.log('\n📋 可用模型:');
                models.forEach((m, i) => console.log(`  ${i + 1}. ${m}`));
                console.log();
                askQuestion();
                return;
            }

            if (input.startsWith('/model ')) {
                currentModel = input.slice(7);
                console.log(`✅ 已切换到: ${currentModel}\n`);
                askQuestion();
                return;
            }

            console.log(`🤖 ${currentModel}:`);
            const answer = await ai.chat(currentModel, input);
            console.log(`   ${answer}\n`);

            askQuestion();
        });
    };

    askQuestion();
}

// 运行示例
if (require.main === module) {
    // main(); // 批量测试
    interactiveMode(); // 交互式模式
}

module.exports = MultiModelAI;

File: .claude/examples/model_switcher.py
#!/usr/bin/env python3
"""
AI 模型切换使用示例
支持本地 cursorweb2api 的 24 个模型
"""

import requests
import json

class MultiModelAI:
    def __init__(self, base_url="http://localhost:8000/v1", api_key="aaa"):
        self.base_url = base_url
        self.api_key = api_key

    def chat(self, model, message, **kwargs):
        """
        调用指定模型进行对话

        Args:
            model: 模型ID（如 "claude-4.5-sonnet", "gpt-4o" 等）
            message: 用户消息
            **kwargs: 其他参数（temperature, max_tokens 等）
        """
        url = f"{self.base_url}/chat/completions"
        headers = {
            "Content-Type": "application/json",
            "Authorization": f"Bearer {self.api_key}"
        }

        data = {
            "model": model,
            "messages": [{"role": "user", "content": message}],
            **kwargs
        }

        response = requests.post(url, headers=headers, json=data)
        result = response.json()

        return result["choices"][0]["message"]["content"]

    def list_models(self):
        """列出所有可用模型"""
        url = f"{self.base_url}/models"
        headers = {"Authorization": f"Bearer {self.api_key}"}

        response = requests.get(url, headers=headers)
        models = response.json()

        return [model["id"] for model in models["data"]]


# 使用示例
if __name__ == "__main__":
    ai = MultiModelAI()

    # 1. 查看所有可用模型
    print("📋 可用模型:")
    models = ai.list_models()
    for i, model in enumerate(models, 1):
        print(f"  {i}. {model}")

    print("\n" + "="*50 + "\n")

    # 2. 使用 Claude 4.5 Sonnet
    print("🤖 Claude 4.5 Sonnet:")
    response = ai.chat("claude-4.5-sonnet", "用一句话介绍你自己")
    print(f"   {response}\n")

    # 3. 使用 GPT-4o
    print("🤖 GPT-4o:")
    response = ai.chat("gpt-4o", "用一句话介绍你自己")
    print(f"   {response}\n")

    # 4. 使用 Gemini 2.5 Flash（快速）
    print("🤖 Gemini 2.5 Flash:")
    response = ai.chat("gemini-2.5-flash", "用一句话介绍你自己")
    print(f"   {response}\n")

    # 5. 使用 DeepSeek R1（推理）
    print("🤖 DeepSeek R1 (推理):")
    response = ai.chat("deepseek-r1", "1+1=?", temperature=0.1)
    print(f"   {response}\n")

    # 6. 使用 Code Supernova（编程）
    print("🤖 Code Supernova (编程):")
    response = ai.chat("code-supernova-1-million", "写一个Python快速排序函数", max_tokens=500)
    print(f"   {response}\n")

    # 7. 对比不同模型的回答
    print("\n" + "="*50)
    print("📊 对比测试：同一个问题用不同模型")
    print("="*50 + "\n")

    question = "什么是量子纠缠？用一句话解释。"
    test_models = ["claude-4.5-sonnet", "gpt-4o", "gemini-2.5-flash", "deepseek-v3.1"]

    for model in test_models:
        print(f"🤖 {model}:")
        response = ai.chat(model, question, max_tokens=100)
        print(f"   {response}\n")
